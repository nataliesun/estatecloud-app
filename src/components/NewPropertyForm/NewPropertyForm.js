import React, { Component } from 'react';
import DateRangePicker from '@wojtekmaj/react-daterange-picker'
import PropertyContext from '../../contexts/PropertyContext';

import './NewPropertyForm.scss'

class NewPropertyForm extends Component {
  static contextType = PropertyContext

  state = {
    date: [new Date(), new Date()],
  }

  onDateChange = date => this.setState({ date })

  handleSubmit = ev => {
    ev.preventDefault()

    const { address, city, state, status, rent_price, initial_price, mortgage_payment } = ev.target;

    const newProperty = {
      address: address.value,
      city: city.value,
      state: state.value,
      status: status.value,
      rent_price: rent_price.value,
      initial_price: initial_price.value,
      mortgage_payment: mortgage_payment.value
    }

    this.context.addProperty(newProperty)

    this.props.history.push('/dashboard')

  }

  render() {
    return (
      <form className="NewPropertyForm" onSubmit={this.handleSubmit}>
        <div className="input-section">
          <label htmlFor="address">Address: </label>
          <input type="text" name="address" placeholder="321 Awesome Ave." required />
          <label htmlFor="city">City: </label>
          <input type="text" name="city" placeholder="San Diego" required />
          <label htmlFor="state">State: </label>
          <input type="text" name="state" placeholder="CA" required />
        </div>


        <div className="input-section">
          <label htmlFor="available">Available</label>
          <input
            type="radio"
            value="available"
            id="available"
            name="status"
          />
          <label htmlFor="rented">Rented</label>
          <input
            type="radio"
            value="rented"
            id="rented"
            name="status"
          />
          <label htmlFor="occupied">Occupied</label>
          <input
            type="radio"
            value="occupied"
            id="occupied"
            name="status"
          />
        </div>

        <div className="date-range">
          <DateRangePicker
            onChange={this.onDateChange}
            value={this.state.date}
          />
        </div>

        <div className="input-section">
          <label htmlFor="rent-price">Rent price:</label>
          <input type="number" name="rent_price" placeholder="2000" />


          <label htmlFor="intial-price">Initial price:</label>
          <input type="number" name="initial_price" placeholder="1000000" />


          <label htmlFor="mortgage-payment">Mortgage payment: </label>
          <input type="number" name="mortgage_payment" placeholder="1500" />
        </div>

        <div className="submit-buttons">

          <button type="submit">Submit</button>
          <button type="reset">Reset</button>
        </div>
      </form >
    );
  }
}

export default NewPropertyForm