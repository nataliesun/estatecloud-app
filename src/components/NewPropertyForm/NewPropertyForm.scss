@import '../variables.scss';

.NewPropertyForm {
  padding: 0 20px;

  input {
    // border: 1px grey solid;
  }

  

  .input-section{
    display: flex;
    flex-direction: column;
    margin-bottom: 20px;

    label {
      margin-bottom: 10px;
      margin-top: 20px;
    }

    input[type=text], input[type=number] {
      @include big-input;
      
      &::placeholder {
        color: rgb(133, 133, 133);
      }
    }
  }

  .submit-buttons button {
    @include highlighted-button;
    font-size: 16px;
    border: none;
  }
}