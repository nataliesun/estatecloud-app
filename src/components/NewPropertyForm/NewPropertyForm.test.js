import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import NewPropertyForm from './NewPropertyForm';

describe(`NewPropertyForm Component`, () => {

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <NewPropertyForm />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NewPropertyForm />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})