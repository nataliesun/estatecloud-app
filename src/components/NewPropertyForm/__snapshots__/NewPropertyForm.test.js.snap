// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPropertyForm Component renders the UI as expected 1`] = `
<form
  className="NewPropertyForm"
  onSubmit={[Function]}
>
  <div
    className="input-section"
  >
    <label
      htmlFor="address"
    >
      Address: 
    </label>
    <input
      name="address"
      placeholder="321 Awesome Ave."
      required={true}
      type="text"
    />
    <label
      htmlFor="city"
    >
      City: 
    </label>
    <input
      name="city"
      placeholder="San Diego"
      required={true}
      type="text"
    />
    <label
      htmlFor="state"
    >
      State: 
    </label>
    <input
      name="state"
      placeholder="CA"
      required={true}
      type="text"
    />
  </div>
  <div
    className="input-section"
  >
    <label
      htmlFor="available"
    >
      Available
    </label>
    <input
      id="available"
      name="status"
      type="radio"
      value="available"
    />
    <label
      htmlFor="rented"
    >
      Rented
    </label>
    <input
      id="rented"
      name="status"
      type="radio"
      value="rented"
    />
    <label
      htmlFor="occupied"
    >
      Occupied
    </label>
    <input
      id="occupied"
      name="status"
      type="radio"
      value="occupied"
    />
  </div>
  <div
    className="date-range"
  />
  <div
    className="input-section"
  >
    <label
      htmlFor="rent-price"
    >
      Rent price:
    </label>
    <input
      name="rent_price"
      placeholder="2000"
      type="number"
    />
    <label
      htmlFor="intial-price"
    >
      Initial price:
    </label>
    <input
      name="initial_price"
      placeholder="1000000"
      type="number"
    />
    <label
      htmlFor="mortgage-payment"
    >
      Mortgage payment: 
    </label>
    <input
      name="mortgage_payment"
      placeholder="1500"
      type="number"
    />
  </div>
  <div
    className="submit-buttons"
  >
    <button
      type="submit"
    >
      Submit
    </button>
    <button
      type="reset"
    >
      Reset
    </button>
  </div>
</form>
`;
